janitor/functions/case_when.py

```diff
@@ -99,100 +114,126 @@ def case_when(df: pd.DataFrame, *args, column_name: str) -> pd.DataFrame:
         `result` can be a scalar, a 1-D array, or a callable.
         If `result` is a callable, it should evaluate to a 1-D array.
         For a 1-D array, it should have the same length as the DataFrame.
-        The `default` argument applies if none of `condition0`,
-        `condition1`, ..., evaluates to `True`.
-        Value can be a scalar, a callable, or a 1-D array. if `default` is a
-        callable, it should evaluate to a 1-D array.
+    :param default: scalar, 1-D array or callable.
+        If callable, it should evaluate to a 1-D array.
         The 1-D array should be the same length as the DataFrame.
+        The element inserted in the output when all conditions
+        evaluate to False. Default is 0.
     :param column_name: Name of column to assign results to. A new column
         is created, if it does not already exist in the DataFrame.
-    :raises ValueError: If the condition fails to evaluate.
     :returns: A pandas DataFrame.
     """
-    conditions, targets, default = _case_when_checks(df, args, column_name)
+    check("column_name", column_name, [str])
+    default = if_else(df, args, default)
+    return df.assign(**{column_name: default})
+
 
-    if len(conditions) == 1:
-        default = default.mask(conditions[0], targets[0])
-        return df.assign(**{column_name: default})
+def if_else(df: pd.DataFrame, args, default: Optional[Any] = 0) -> pd.Series:
```

Comment:
@samukweku this is a question more than a critique -- what would make more semantic sense for the default value of the `default` argument? Should it be None/`np.nan` instead of 0? What are your thoughts here? I think you might have given more thought to this problem that I have, so I'm happy to defer to you here.

Reply:
I am open to criticism :grin: ... I just chose `numpy.select` default value ... thinking about it I'd say None is a better fit. open to suggestions

Reply:
I'll edit and let the user pass a default value instead